pyanalyze/implementation.py

```diff
@@ -416,6 +416,14 @@ def _dict_setitem_impl(ctx: CallContext) -> ImplReturn:
     return ImplReturn(KnownValue(None))
 
 
+def _list_getitem_impl(ctx: CallContext) -> ImplReturn:
+    self_value = ctx.vars['self']
+    index = ctx.vars['index']
+    if isinstance(index, TypedValue):
+        if index.typ is slice:
+            return TypedValue(self_value.get_type())
+    return UNRESOLVED_VALUE
```

Comment:
This should return at least the type parameter of the list, so that if you do `lst[0]` on a `List[int]`, you at least get `int` back.